core/src/voice/envelopes.rs

```diff
@@ -219,14 +266,33 @@ impl EnvelopeDescriptor {
     pub fn to_envelope_params(
         &self,
         samplerate: u32,
-        options: SoundfontInitOptions,
+        options: EnvelopeOptions,
     ) -> EnvelopeParameters {
         let samplerate = samplerate as f32;
 
-        let release = if options.linear_release {
-            EnvelopePart::lerp(0.0, (self.release * samplerate) as u32)
-        } else {
-            EnvelopePart::lerp_to_zero_curve((self.release * samplerate) as u32)
+        // The following are in dB scale (or cents for modulation) so:
+        // Linear dB -> Concave or Convex in amp
+        // Concave or Convex in dB -> Linear in amp
+
+        let attack = match options.attack_curve {
+            EnvelopeCurveType::Linear => {
+                EnvelopePart::lerp_convex(1.0, (self.attack * samplerate) as u32)
+            }
+            _ => EnvelopePart::lerp(1.0, (self.attack * samplerate) as u32),
+        };
+
+        let decay = match options.decay_curve {
+            EnvelopeCurveType::Concave => {
+                EnvelopePart::lerp(self.sustain_percent, (self.decay * samplerate) as u32)
+            }
+            _ => EnvelopePart::lerp_concave(self.sustain_percent, (self.decay * samplerate) as u32),
```

Comment:
I'm not sure what's happening here, it's

If concave: lerp
If linear or convex: concave

Same with the below one

I noticed you also had `unreachable!()` for unsupported sections below, imo it's better to either make being in an unreachable state impossible via types, or just supporting it as-is

Reply:
Yeah I had left a comment about this right above this code. Both the SFZ and SF2 specs define envelope curves in dB, and when converting dB to amplitude it is a 10^x function so linear in dB -> curved in amp and curved in dB -> linear in amplitude.

Also normally synths also this exponential function to calculate their envelope curves but it would be very slow here (and maybe not even possible with the simd lerper) so we did a x^8 which gives pretty similar curves.